### DISCUSSION TITLE
EBU R128 Normalization

### DISCUSSION THREAD

**Discussion by aetx** [PaulWoitaschek/Voice#1951] [Ideas]
Some audiobooks have a rather large dynamic range, so that they are difficult to use while external noise exists. This can be a problem for listening to audiobooks on the move, such as in a car or a bus. Quiet parts are then overshadowed by external noise.

Applying normalization removes this problem. In particular I am thinking of the windowed calculation of the EBU R128 loudness as reference. If you are familiar with ReplayGain, this is similar, but instead of a fixed gain per file, the gain changes during the audiobook.

This has been implemented in the windows audio player foobar2000 in the [foo_r128norm](https://www.foobar2000.org/components/view/foo_r128norm) component. I would be happy to preprocess my audiobooks using that prior to listening, but due to the lossy encoding, I would like to avoid re-encoding the files.

edit: This would also solve #1712

**Comment by moddx** [comment]
I want this so much. Might have to look at how the foobar plugin does it. Probably not easy to adapt within android, but I'll try to look at it on the weekend.

**Reply by PaulWoitaschek** [reply]
Search on github for "exoplayer R128". Also check if this is supported by ffmpeg and if it's possible with the exoplayer ffmpeg integration
